contracts/peripheral/IncurDebt.sol

```diff
@@ -443,60 +430,67 @@ contract IncurDebt is OlympusAccessControlledV2, IIncurDebt {
         Borrower storage borrower = borrowers[msg.sender];
 
         if (borrower.debt == 0) revert IncurDebt_BorrowerHasNoOutstandingDebt(msg.sender);
-        IERC20(OHM).safeTransferFrom(msg.sender, address(this), _ohmAmount);
 
         totalOutstandingGlobalDebt -= _ohmAmount;
-        borrower.debt = uint128(borrower.debt - _ohmAmount);
-
+        borrower.debt -= uint128(_ohmAmount);
+                
+        IERC20(OHM).safeTransferFrom(msg.sender, address(this), _ohmAmount);
         ITreasury(treasury).repayDebtWithOHM(_ohmAmount);
+
         emit DebtPaidWithOHM(msg.sender, _ohmAmount, borrower.debt, totalOutstandingGlobalDebt);
     }
 
     /// @notice gets available OHM to borrow for account
     /// @return amount OHM available to borrow
     function getAvailableToBorrow() public view returns (uint256) {
-        uint256 _gonsPerFragment = TOTAL_GONS / IsOHM(sOHM).totalSupply();
-        uint256 sBalance = uint128(borrowers[msg.sender].borrowerGonBalance / _gonsPerFragment);
-        return sBalance - borrowers[msg.sender].debt;
+        uint256 ohmBalance = IgOHM(gOHM).balanceFrom(borrowers[msg.sender].collateralInGOHM);
+        return ohmBalance - borrowers[msg.sender].debt;
+    }
+
+    function _borrow(uint256 _ohmAmount) internal {
+        Borrower storage borrower = borrowers[msg.sender];
+
+        if (_ohmAmount > borrower.limit - borrower.debt) revert IncurDebt_AboveBorrowersDebtLimit(_ohmAmount);
+        if (_ohmAmount > getAvailableToBorrow()) revert IncurDebt_OHMAmountMoreThanAvailableLoan(_ohmAmount);
+
+        borrower.debt += uint128(_ohmAmount);
+        totalOutstandingGlobalDebt += _ohmAmount;
+
+        uint256 totalDebtInGOHM = IgOHM(gOHM).balanceTo(borrower.debt);
+
+        if (totalDebtInGOHM > borrower.unwrappedGOHM) { //is there issue comparing uint256 to uint128
+            uint128 amountToUnwrap = borrower.collateralInGOHM - borrower.unwrappedGOHM;
+            borrower.unwrappedGOHM += amountToUnwrap;
+            IStaking(staking).unwrap(address(this), amountToUnwrap); // Due to rounding error should have tiny less sOhm in wallet than amount they borrowed. Keep small sOhm amount in wallet to make sure no errors.
+        }
+
+        ITreasury(treasury).incurDebt(_ohmAmount, OHM);
     }
 
     /// @notice pays borrower debit and return left collateral and paid debt
     /// @param _borrower the account debit needs to be paid
-    /// @return currentCollateral left collateral after paying debit
+    /// @return collateralRemaining left collateral after paying debit
     /// @return paidDebt debt paid
     function _repay(address _borrower)
```

Comment:
Kindly add something of this sort

```
function _repay(address _borrower, bool _seize)………
if(!_seize){
	uint256 sOHMToWrap = IgOHM(gOHM).balanceFrom(borrower.unwrappedGOHM) - debt;
	borrower.unwrappedGOHM = 0;
	IStaking(staking).wrap(address(this), sOHMToWrap); // wrap up remaining sOHM into gOHM
}
```

Reply:
So for seize im unstaking the gOHM to OHM to burn anyway. If we do this and repay doesn't wrap up excess sOHM we can run into situation where we have some collateral in sOHM some in gOHM which makes burning bit more difficult logic. So rather just wrap up all excess sOHM and then in burn function unstake and burn it. Also cleaner for the repay function.